tests/unit/test_symmetric_crypto.cpp

```diff
@@ -0,0 +1,312 @@
+}
+
+TEST_P(SymmetricCipherAdvancedTest, readDecryptedTextInChunks)
+{
+    auto operationMode = GetParam();
+
+    // Given: long encrypted message
+    auto encryptoBuilder = AESCipherBuilder{operationMode, _secretKey};
+    std::shared_ptr<SymmetricCipher> encryptor;
+    if (isAuthenticatedCipherMode(operationMode)) {
+        encryptor = encryptoBuilder.buildAuthenticatedEncryptor();
+    }
+    else {
+        encryptor = encryptoBuilder.buildEncryptor();
+    }
+
+    encryptor->update(_plaintext);
+    auto ciphertext = encryptor->finish();
+    auto iv = encryptor->getIV();
+    std::vector<uint8_t> tag;
+    auto authenticatedEncryptor = dynamic_cast<AuthenticatedEncryptionI*>(encryptor.get());
+    if (authenticatedEncryptor) {
+        tag = authenticatedEncryptor->getAuthTag();
+    }
+
+    // When: decrypt in chunks
+    auto decryptorBuilder = AESCipherBuilder{operationMode, _secretKey}.setIV(iv);
+    std::shared_ptr<SymmetricCipher> decryptor;
+    if (isAuthenticatedCipherMode(operationMode)) {
+        decryptor = encryptoBuilder.buildAuthenticatedDecryptor();
+    }
+    else {
+        decryptor = encryptoBuilder.buildDecryptor();
+    }
+
+    std::vector<uint8_t> decryptedText;
+
+    // Decrypt packets and read decrypted text in chunks. Size of these chunks is different.
+    static int const DecryptionChunkSize = 1024 * 4;
+    static int const ReaderBufferSize = 1024;
+    auto ciphertextRunner = ciphertext.begin();
+    while (std::distance(ciphertextRunner, std::end(ciphertext)) > DecryptionChunkSize) {
+        decryptor->update({ciphertextRunner, ciphertextRunner += DecryptionChunkSize});
+        auto decryptedChunk = decryptor->read(ReaderBufferSize);
+        decryptedText.insert(std::end(decryptedText), std::begin(decryptedChunk), std::end(decryptedChunk));
+    }
+    // Decrypt the rest which didn't fit into decryption chunk
+    decryptor->update({ciphertextRunner, std::end(ciphertext)});
+
+    auto authenticatedDecryptor = dynamic_cast<AuthenticatedEncryptionI*>(decryptor.get());
+    if (authenticatedDecryptor) {
+        authenticatedDecryptor->setAuthTag(tag);
+    }
+    auto decryptedChunkFinal = decryptor->finish();
+    decryptedText.insert(std::end(decryptedText), std::begin(decryptedChunkFinal), std::end(decryptedChunkFinal));
+
+    ASSERT_THAT(decryptedText, ::testing::ElementsAreArray(_plaintext));
+}
+
+INSTANTIATE_TEST_SUITE_P(Chunks,
+                         SymmetricCipherAdvancedTest,
+                         testing::ValuesIn(AllSupportedCIpherModesToTest));
+
+
+
+
+class SymmetricCipherMemoryStrategy : public  ::testing::Test {
+protected:
+    void SetUp() override {
+        for (int i = 0; i < kNumberOfTestBlocks; i++) {
+            std::vector<uint8_t> block(kTestBlockLength, i);
+            _expectedData.insert(std::end(_expectedData), std::begin(block), std::end(block));
+        }
+    }
+
+    QueueOfVectorsMemoryStrategy sut;
+
+    const int kNumberOfTestBlocks = 4;
```

Comment:
we usually don't prefix constants with k but rather have all caps with underscope in this lib

Reply:
done